src/Compilers/CSharp/Test/Semantic/Semantics/TargetTypedObjectCreationTests.cs

```diff
@@ -4210,6 +4210,51 @@ public void M2(C c)
                 );
         }
 
+        [Fact]
+        public void TestInConstructorOverloadWithUseSiteError()
+        {
+            var missing = @"public class Missing { }";
+            var missingComp = CreateCompilation(missing, assemblyName: "missing");
+
+            var lib = @"
+public class C
+{
+    public C(Missing m) => throw null;
+    public C(D d) => throw null;
+}
+public class D { }
+";
+            var libComp = CreateCompilation(lib, references: new[] { missingComp.EmitToImageReference() });
+
+            var source = @"
+class D
+{
+    public void M()
+    {
+        new C(new());
+        new C(default);
+        new C(null);
+        C c = new(null);
```

Comment:
>C c = new(null); [](start = 8, length = 16)

Please add a test for a target typed new as an argument of a call that would assert use-site errors caused by the target typed new.